store/store.go

```diff
@@ -38,6 +38,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/juju/ratelimit"
```

Comment:
I looked at this one when investigating econnreset issues. IIRC juju/ratelimit is nicely packaged for all our targets already so it's a reasonable choice.

Reply:
I tend to resist adding a new dependency for such a small amount of code. Hopefully it won't be a nuisance to distribute across all platforms we need it indeed.